Remove process_group_agent and faulty_process_group_agent files

H-Huang opened the pull request:
Remove the process_group_agent and faulty_process_group_agent now that PROCESS_GROUP backend has been deprecated for RPC. Discussed that it was okay to remove ProcessGroupAgentTest and ProcessGroupAgentBench which depended on process_group_agent from internal code.

Differential Revision: [D30195576](https://www.internalfb.com/diff/D30195576)

pritamdamania87 commented:
Looks like there are a bunch of test failures. Also, can we remove `test_process_group_agent.py` from https://github.com/pytorch/pytorch/blob/master/torch/distributed/CONTRIBUTING.md and probably add an example with tensorpipe?

H-Huang commented:
Thanks @pritamdamania87, updated the PR to fix the test failures. I will also update our team docs and a few code comments that mention ProcessGroupAgent in a separate PR

H-Huang committed:
Remove process_group_agent and faulty_process_group_agent files (#62985)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/62985

Remove the process_group_agent and faulty_process_group_agent now that PROCESS_GROUP backend has been deprecated for RPC. Discussed with xush6528 that it was okay to remove ProcessGroupAgentTest and ProcessGroupAgentBench which depended on process_group_agent.

Test Plan: CI tests

Differential Revision: D30195576

fbshipit-source-id: 1ea8568f6120f0f02ef578935948c5a65e01f586

mrshenli reviewed (approved):
Thanks for removing ProcessGroupAgent!

nit and can be done in followup PRs: there are some vestige still mentions ProcessGroupAgent, below are some examples:

https://github.com/pytorch/pytorch/blob/ea808df25deee98a3021f5cff1f818803d56c97b/torch/testing/_internal/distributed/rpc/faulty_rpc_agent_test_fixture.py#L53-L54

https://github.com/pytorch/pytorch/blob/ea808df25deee98a3021f5cff1f818803d56c97b/torch/csrc/distributed/rpc/message.h#L104

https://github.com/pytorch/pytorch/blob/2006dc63168963d9ea607a54c8eb7abe7dc2a486/torch/testing/_internal/distributed/rpc/dist_autograd_test.py#L1467

https://github.com/pytorch/pytorch/blob/ea808df25deee98a3021f5cff1f818803d56c97b/torch/testing/_internal/distributed/rpc/rpc_test.py#L2886
